app/Services/PurchaseOrder/PurchaseOrderService.php

```diff
@@ -75,6 +77,32 @@ public function getPurchaseOrderPdf($contact = null)
         return (new GetPurchaseOrderPdf($this->purchase_order, $contact))->run();
     }
 
+    public function getEPurchaseOrder($contact = null)
+    {
+        return (new CreateEDocument($this->purchase_order))->handle();
+    }
+    public function deleteEPurchaseOrder()
+    {
+        $this->purchase_order->load('invitations');
+
+        $this->purchase_order->invitations->each(function ($invitation) {
+            try {
+                // if (Storage::disk(config('filesystems.default'))->exists($this->invoice->client->e_invoice_filepath($invitation).$this->invoice->getFileName("xml"))) {
+                Storage::disk(config('filesystems.default'))->delete($this->purchase_order->client->e_document_filepath($invitation).$this->purchase_order->getFileName("xml"));
```

Comment:
->contact->vendor

Reply:
Unsure about this one. Changed it, but not sure it how you intended